### PR TITLE
[OpenSSL] support for "no-autoload-config" config option

### PR DESCRIPTION
**Pull request by lstipakov**
OpenSSL has "no-autoload-config" config option, which disables automatic config loading:

  https://github.com/openssl/openssl/pull/5959

This patch allows to set above option via variable in custom triplet:

  \#build-openssl-no-autoload-config.cmake
  `set(OPENSSL_NO_AUTOLOAD_CONFIG ON)`

and use it like:

  `vcpkg install --overlay-triplets=<path-to-triplets> openssl:build-openssl-no-autoload-config`

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@lstipakov ,Thanks for your pr.
Please run the command "./vcpkg x-add-version openssl --overwrite-version" then commit the change

**Comment by lstipakov** [comment]
> Please run the command "./vcpkg x-add-version openssl --overwrite-version" then commit the change

Done!

**Comment by JonLiu1993** [comment]
@lstipakov ,Thanks for your pr .
Have you tested that the added option can be used normally？

**Comment by lstipakov** [comment]
Without this option I see with procmon that my app tries to load `openssl.cnf` from certain location on disk, for example `c:\vcpkg\packages\openssl_x64-windows\openvpn.cnf`. Then I created triplet where this option is set: `set(OPENSSL_NO_AUTOLOAD_CONFIG ON)`. When my app uses openssl which is built with this triplet, that config is not read anymore. Does it answer your question?

**Comment by JackBoosY** [comment]
Can you double confirm that this feature is applicable to Openssl 1.1.1k in vcpkg?

**Comment by lstipakov** [comment]
Yes, I have been using OpenSSL 1.1.1k, which has been in vcpkg for a few month already (https://github.com/microsoft/vcpkg/pull/16937)

File: ports/openssl/windows/portfile.cmake [hunk 1]
```diff
@@ -29,6 +29,10 @@ if(DEFINED OPENSSL_USE_NOPINSHARED)
     set(CONFIGURE_OPTIONS ${CONFIGURE_OPTIONS} no-pinshared)
 endif()
 
+if(DEFINED OPENSSL_NO_AUTOLOAD_CONFIG)
+    set(CONFIGURE_OPTIONS ${CONFIGURE_OPTIONS} no-autoload-config)
```
Comment (JackBoosY) [inline ports/openssl/windows/portfile.cmake:33]: `no-autoload-config` exists on the current version in vcpkg.

**Review by JackBoosY — APPROVED** [review]

File: ports/openssl/windows/portfile.cmake [hunk 2]
```diff
@@ -29,6 +29,10 @@ if(DEFINED OPENSSL_USE_NOPINSHARED)
     set(CONFIGURE_OPTIONS ${CONFIGURE_OPTIONS} no-pinshared)
 endif()
 
+if(DEFINED OPENSSL_NO_AUTOLOAD_CONFIG)
```
Comment (ras0219-msft) [inline ports/openssl/windows/portfile.cmake:32]:
```suggestion
if(OPENSSL_NO_AUTOLOAD_CONFIG)
```
In addition to testing if it's defined, we should also ensure it is a truthy value.

**Review by ras0219-msft — APPROVED** [review]
This LGTM once my suggestion is applied. Thanks for the PR!

File: ports/openssl/windows/portfile.cmake (hunk 2 above)
Comment (lstipakov) [reply ports/openssl/windows/portfile.cmake]: Done. Originally I copied it from the previous block which has `if(DEFINED OPENSSL_USE_NOPINSHARED)`.

**Comment by lstipakov** [comment]
Hi, is there anything else needed to be done here before it gets merged?

**Comment by JackBoosY** [comment]
Ping @ras0219 @ras0219-msft for merge this PR.

**Comment by ras0219-msft** [comment]
Thanks for the PR!

**Pull request merged**